### PR TITLE
[MM-16688] Make DMs permissions aware

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Checking for view issue permissions.
- Unless it's a comment, then check for permissions to view the comment.

#### Link
[MM-16688](https://mattermost.atlassian.net/browse/MM-16688)

### CODE REVIEW

File: server/webhook.go [hunk 1]
```diff
@@ -101,6 +102,46 @@ func (wh *webhook) PostNotifications(p *Plugin) ([]*model.Post, int, error) {
...
+			continue
+		}
+		jiraClient, err2 := ji.GetJIRAClient(jiraUser)
+		if err2 != nil {
+			p.errorf("PostNotifications: error while getting jiraClient, err: %v", err2)
+			continue
+		}
+		// If this is a comment-related webhook, we need to check if they have permissions to read that.
+		// Otherwise, check if they can view the issue.
+		if wh.eventMask&maskComments != 0 {
```
Comment (levb) [inline server/webhook.go]: Haha, the bitmask comes convenient!

File: server/webhook.go [hunk 2]
```diff
@@ -101,6 +102,46 @@ func (wh *webhook) PostNotifications(p *Plugin) ([]*model.Post, int, error) {
 			continue
 		}
 
+		// Check if the user has permissions.
```
Comment (levb) [inline server/webhook.go]: maybe add a label to the for loop, so it becomes `continue NextNotification`?

Comment (cpoile) [reply server/webhook.go]: Something like:
image

Comment (levb) [reply server/webhook.go]: Exactly

**Pull request merged**